test/320_claim_3_test.sh

```diff
@@ -52,4 +52,22 @@ weave_on $HOST3 prime
 stop_weave_on $HOST2
 weave_on $HOST1 launch-router $HOST3
 
+stop_weave_on $HOST1
+stop_weave_on $HOST2
+stop_weave_on $HOST3
+delete_persistence $HOST1 $HOST2 $HOST3
+
+weave_on $HOST1 launch $HOST2
+# Start another container on host1. The starting should block, because host1 is
+# not able to establish the ring due to host2 or host3 being offline.
+CMD="proxy_start_container $HOST1 --name c4 -e WEAVE_CIDR=10.32.0.4/12"
+assert_raises "timeout 5 cat <( $CMD )"  124
+# However, allocation for an external subnet should not block.
+assert_raises "proxy_start_container $HOST1 -e WEAVE_CIDR=10.48.0.1/12"
+
+# Launch host2, so that host1 could establish the ring.
+weave_on $HOST2 launch $HOST1
+sleep 2
```

Comment:
I reckon if you ran `weave_on $HOST1 prime` here you could get rid of the `sleep`.